android-interop-testing/app/src/main/java/io/grpc/android/integrationtest/TesterActivity.java

```diff
@@ -0,0 +1,137 @@
+ * met:
+ *
+ *    * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *    * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.grpc.android.integrationtest;
+
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.text.TextUtils;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.google.android.gms.security.ProviderInstaller;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class TesterActivity extends AppCompatActivity
+    implements ProviderInstaller.ProviderInstallListener {
+  private List<Button> buttons;
+  private EditText hostEdit;
+  private EditText portEdit;
+  private TextView resultText;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_tester);
+    buttons = new LinkedList<>();
+    buttons.add((Button) findViewById(R.id.empty_unary_button));
+    buttons.add((Button) findViewById(R.id.large_unary_button));
+    buttons.add((Button) findViewById(R.id.client_streaming_button));
+    buttons.add((Button) findViewById(R.id.server_streaming_button));
+    buttons.add((Button) findViewById(R.id.ping_pong_button));
+
+    hostEdit = (EditText) findViewById(R.id.host_edit_text);
+    portEdit = (EditText) findViewById(R.id.port_edit_text);
+    resultText = (TextView) findViewById(R.id.grpc_response_text);
+
+    InteropTester.initTestCa(getResources().openRawResource(R.raw.ca));
+    ProviderInstaller.installIfNeededAsync(this, this);
```

Comment:
This isn't enough when using the TesterActivity. Nothing prevents the installation from not having occurred in time for usage.